tools/webhook-apicoverage/resourcetree/test_util.go

```diff
@@ -0,0 +1,152 @@
+}
+
+type combinedNodeType struct {
+	b baseType
+	a arrayType
+	p ptrType
+}
+
+func verifyBaseTypeNode(data nodeData) error {
+	if len(data.children) != 2 {
+		return errors.New(fmt.Sprint("BaseTypeNodeVerification: Expected 2 children got only : ", len(data.children)))
+	}
+
+	if value, ok := data.children["field1"]; ok {
+		n := value.getData()
+		if !n.leafNode || n.fieldType.Kind() != reflect.String || n.fieldType.PkgPath() != "" || len(n.children) != 0 {
+			return errors.New("BaseTypeNodeVerification: Invalid Child Node for field1")
+		}
+	} else {
+		return errors.New("BaseTypeNodeVerification: field1 Child Not found")
+	}
+
+	return nil
+}
+
+func verifyPtrNode(data nodeData) error {
+	if len(data.children) != 2 {
+		return fmt.Errorf("PtrNodeVerification: Expected 2 children got: %d", len(data.children))
+	}
+
+	child := data.children["structPtr"]
+	if len(child.getData().children) != 1 {
+		return errors.New("PtrNodeVerification: Unexpected field type for field:structPtr")
+	}
+
+	child = child.getData().children["structPtr-ptr"]
+	if err := verifyBaseTypeNode(child.getData()); err != nil {
+		return fmt.Errorf("PtrNodeVerification: Invalid Ptr type found for field:structPtr-ptr - %v", err)
+	}
+
+	child = data.children["basePtr"]
+	if len(child.getData().children) != 1 {
+		return errors.New("PtrNodeVerification: Unexpected field type for field:basePtr")
+	}
+
+	child = child.getData().children["basePtr-ptr"]
+	if child.getData().fieldType.Kind() != reflect.Float32 || !child.getData().leafNode || child.getData().fieldType.PkgPath() != "" || len(child.getData().children) != 0 {
+		return errors.New("PtrNodeVerification: Unexpected field type for field:basePtr-ptr")
+	}
+
+	return nil
+}
+
+func verifyArrayNode(data nodeData) error {
+	if len(data.children) != 2 {
+		return fmt.Errorf("ArrNodeVerification: Expected 2 children got: %d", len(data.children))
+	}
+
+	child := data.children["structArr"]
+	if child.getData().fieldType.Kind() != reflect.Slice || len(child.getData().children) != 1 {
+		return errors.New("ArrNodeVerification: Unexpected field type for field:structArr")
+	}
+
+	child = child.getData().children["structArr-arr"]
+	if err := verifyBaseTypeNode(child.getData()); err != nil {
+		return fmt.Errorf("ArrNodeVerification: Invalid ArrorSlice type found for field:structArr-arr - %v", err)
+	}
+
+	child = data.children["baseArr"]
+	if child.getData().fieldType.Kind() != reflect.Slice || len(child.getData().children) != 1 {
+		return errors.New("ArrNodeVerification: Unexpected field type for field:baseArr")
+	}
+
+	child = child.getData().children["baseArr-arr"]
+	if child.getData().fieldType.Kind() != reflect.Bool || !child.getData().leafNode || child.getData().fieldType.PkgPath() != "" || len(child.getData().children) != 0 {
+		return errors.New("ArrNodeVerification: Unexpected field type for field:baseArr-arr")
+	}
+
+	return nil
+}
+
+func verifyOtherTypeNode(data nodeData) error {
+	if len(data.children) != 2 {
```

Comment:
feels like we are doing the same thing in every verify method. Can we combine these into a single method and every verifyFoo calls that instead?

Reply:
Each type is different, the uber verification would have to take parameters specific to each test. Any addition of new verification logic would require changing that method and all other tests. whereever it is possible I have moved it to a common logic.